bytearray_operations.go

```diff
@@ -44,6 +44,83 @@ func LeftShift(data []byte, shift uint64) []byte {
 	return result
 }
 
+// Mask apply AND mask to byte array
+func Mask(data, mask []byte) []byte {
+	var dataLength = len(data)
+	result := make([]byte, dataLength)
+	if dataLength < 1 {
+		return result
+	}
+	copy(result, data)
+
+	var maskLength = len(mask)
+	var operationLength = maskLength
+	// If mask is longer than data, keep operation to data length
+	if maskLength > dataLength {
+		operationLength = dataLength
+	}
+	for i := 0; i < operationLength; i++ {
```

Comment:
You need to support little endian, not big endian (see issue #4)

Reply:
WIP.

Reply:
Need a little help here. Currently mask applied on data as follows:

`d1 d2 d3 d4`
`m1 m2`

Does little-endian mean that mask applied applied as follows?

`d1 d2 d3 d4`
`      m1 m2`

Reply:
It's more like that 
```
d4  d3  d2  d1
            m2  m1
```
or
```
              d2  d1
m4  m3 m2  m1
```